sonaranalyzer-dotnet/its/expected/Nancy/Nancy.Demo.Hosting.Aspnet-{E127FED3-01C0-41BA-BF83-D8DCDD827D6A}-S1905.json

```diff
@@ -1,17 +0,0 @@
-{
-"issues":  [
-{
-"id":  "S1905",
-"message":  "Remove this unnecessary cast to 'System.Collections.Generic.IEnumerable<MyUberRouteMetadata>'.",
-"location":  {
-"uri":  "sources\Nancy\src\Nancy.Demo.Hosting.Aspnet\MainModule.cs",
-"region":  {
-"startLine":  38,
-"startColumn":  101,
-"endLine":  38,
-"endColumn":  131
-}
-}
-}
-]
-}
```

Comment:
Wasn't this a true positive?
```
.WithModel(routeCacheProvider.GetCache().RetrieveMetadata<MyUberRouteMetadata>().OfType<MyUberRouteMetadata>())
```
In this case `RetrieveMetadata<MyUberRouteMetadata>` will return `IEnumerable<MyUberRouteMetadata>`

Reply:
If `MyUberRouteMetadata` is a reference type and it's not a struct, than `OfType<MyUberRouteMetadata>` filters out `null` values

Reply:
and it's a class
```
public class MyUberRouteMetadata
```

Reply:
Ok, since the IEnumerable could contain null values and there is no way to know for sure if these are present or not, we prefer to check this only for value types (https://github.com/SonarSource/sonar-dotnet/issues/3273#issuecomment-656010838)